### DISCUSSION TITLE
How to fix content security policy for Nextcloud

### DISCUSSION THREAD

**Discussion by jwgn** [tobychui/zoraxy#321] [Q&A]
Since latest Nextcloud update, I am no longer able to install Apps. According [to an issue ](https://github.com/nextcloud/server/issues/44685#issuecomment-2067203058), this is caused by reverse proxy configuration:

`you are using a proxy / reverse proxy that does not correctly rewrites redirects.`

How do we fix this on zoraxy?

**Comment by tobychui** [comment]
@jwgn I see you are struggling using NextCloud with Zoraxy, but unfortunately I don't use NextCloud and therefore there isn't much I can help. Maybe you can post your Zoraxy config here and see if someone can offer you some help regarding this?

**Comment by jwgn** [comment]
Well, I have just set up the host without any special configuration. The comment in the nextcloud issue says:

```
Solution: Ensure redirects are rewritten using the correct protocol on your reverse proxy.
For NGinx e.g. ensure to use [proxy_redirect](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_redirect).
Note: If you use proxy_pass this is the default, EXCEPT you use variables in the proxy_pass.

For other reverse proxies, refer to their documentation.
```

So I am just asking on how to configure this correctly on Zoraxy.

**Comment by jwgn** [comment]
Seems this was a Nextcloud bug. Just installed the 30.0.1 update from today and the issue is gone.

**Comment by jwgn** [comment]
Hm:

`We did not fix the wrong configuration but applied a workaround, by preventing a potential redirect. Yet any other app could rely on this valid bahavior so you could have similar problems in other apps.`

So it seems we still need to fix something on Zoraxy?

**Reply by tobychui** [reply]
@jwgn As I recall there were people getting NextCloud working with Zoraxy. Zoraxy follows (most of the) Apache web server logic for handling redirects. It is not Zoraxy issue for NextCloud not compatible with anything but Nginx for reverse proxy.  

And given that you didn't provide any information regarding your setups and steps for reproducing the issue, there is nothing I can test  for you. As always, if you figure out a workaround, feel free to contribute to the Zoraxy wiki or even better, create a PR with new automatic sniffing logic that applies such workaround 👍🏻
